cudamapper/src/main.cu

```diff
@@ -266,18 +270,28 @@ int main(int argc, char* argv[])
 
                 // Get unfiltered overlaps
                 std::vector<claragenomics::cudamapper::Overlap> overlaps_to_add;
+
                 overlapper.get_overlaps(overlaps_to_add, matcher->anchors(), *query_index, *target_index);
 
-                std::shared_ptr<std::future<void>> write_and_filter_overlaps_future = std::make_shared<std::future<void>>(std::async(std::launch::async,
-                                                                                                                                     [&overlaps_writer_mtx, overlaps_to_add](std::vector<claragenomics::cudamapper::Overlap> overlaps) {
-                                                                                                                                         std::vector<claragenomics::cudamapper::Overlap> filtered_overlaps;
-                                                                                                                                         claragenomics::cudamapper::Overlapper::filter_overlaps(filtered_overlaps, overlaps_to_add);
-                                                                                                                                         std::lock_guard<std::mutex> lck(overlaps_writer_mtx);
-                                                                                                                                         claragenomics::cudamapper::Overlapper::print_paf(filtered_overlaps);
-                                                                                                                                     },
-                                                                                                                                     overlaps_to_add));
+                //Increment counter which tracks number of overlap chunks to be filtered and printed
+                num_overlap_chunks_to_print++;
+                auto print_overlaps = [&overlaps_writer_mtx, &num_overlap_chunks_to_print](std::vector<claragenomics::cudamapper::Overlap> overlaps) {
```

Comment:
I'd suggest renaming this to `filter_and_print_overlaps` until filtering is moved to device